src/config.h

```diff
@@ -38,20 +38,18 @@ class Server;
 namespace Engine {
 class Storage;
 }
-#define PORT_LIMIT 65535
 
-#define SUPERVISED_NONE 0
-#define SUPERVISED_AUTODETECT 1
-#define SUPERVISED_SYSTEMD 2
-#define SUPERVISED_UPSTART 3
+constexpr const uint16_t PORT_LIMIT = 65535;
 
-#define TLS_AUTH_CLIENTS_NO "no"
-#define TLS_AUTH_CLIENTS_OPTIONAL "optional"
+enum SupervisedMode { SUPERVISED_NONE = 0, SUPERVISED_AUTODETECT, SUPERVISED_SYSTEMD, SUPERVISED_UPSTART };
```

Comment:
Now that enum has a name, can the prefix of enum values be removed?

Reply:
I think we cannot remove it. Since it is not an `enum class`, the names in the `enum` will be directly exported to the outer namespace, so if we remove the prefix, we will get some name like `NONE` in the gloabl namespace.

Reply:
And the reason I do not use `enum class` here is that `enum class` cannot be implicitly casted to integer, so it will involve more code changes.

Reply:
Ok, got it!